### PR TITLE
Rust: Setup inline flow test library

### PR DESCRIPTION
**Pull request by paldepind**
Setup inline flow test library for Rust. It doesn't output anything, but it should once other things begin to work 😉

### CODE REVIEW

File: rust/ql/test/utils/InlineFlowTest.qll [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        .getArgList()
+        .getAnArg() = sink.asExpr()
+  }
+
+  private string getSourceArgString(DataFlow::Node src) {
+    defaultSource(src) and
+    result = src.asExpr().(CallExpr).getArgList().getArg(0).toString()
+  }
+
+  string getArgString(DataFlow::Node src, DataFlow::Node sink) {
```
Comment (hvitved) [inline rust/ql/test/utils/InlineFlowTest.qll:29]: Missing `bindingset[src, sink]`.

File: rust/ql/test/utils/InlineFlowTest.qll [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    (if exists(getSourceArgString(src)) then result = getSourceArgString(src) else result = "") and
```
Comment (hvitved) [inline rust/ql/test/utils/InlineFlowTest.qll:30]: I know this was copied from Ruby, but the idiomatic way to write things like this is
```ql
result = getSourceArgString(src)
or
not exists(getSourceArgString(src)) and
result = ""
```

**Review by hvitved — CHANGES_REQUESTED** [review]
Missing a `bindingset`, otherwise LGTM.

File: rust/ql/test/utils/InlineFlowTest.qll (hunk 1 above)
Comment (paldepind) [reply rust/ql/test/utils/InlineFlowTest.qll:30]: What does the `bindingset` do here? From the documentation I thought that it was only to be used for predicates over infinite types. For several of the other language there is no `bindingset` annotation on `getArgString`. [Go for instance](https://github.com/paldepind/codeql/blob/d9d3027667e19e98816c731bcb0dfefec0e0564c/go/ql/test/TestUtilities/InlineFlowTest.qll?plain=1#L24). Should it ideally also be added there? Or is it not needed there for some reason?

Comment (hvitved) [reply rust/ql/test/utils/InlineFlowTest.qll:30]: It is needed to avoid creating a Cartesian product. It should also be added for Go.

**Review by hvitved — APPROVED** [review]

**Pull request merged**